components/views.py

```diff
@@ -5,35 +5,54 @@
 
 from components.filters import ComponentFilter, ComponentPermissionsFilter
 from components.models import Component
-from components.permissions import ComponentPermissions
-from components.serializers import ComponentListSerializer, ComponentSerializer
+from components.permissions import ComponentPermissions, ComponentPrivatePermissions
+from components.serializers import (
+    ComponentControlSerializer,
+    ComponentListSerializer,
+    ComponentSerializer,
+)
 
 
 class ComponentListView(generics.ListCreateAPIView):
     """Use for read-write endpoints to represent a collection of model instances.
     Provides get and post method handlers.
     """
+
     queryset = Component.objects.all()
     serializer_class = ComponentListSerializer
-    permission_classes = [ComponentPermissions, ]
-    filter_backends = [ComponentPermissionsFilter, ]
+    permission_classes = [
+        ComponentPermissions,
+    ]
+    filter_backends = [
+        ComponentPermissionsFilter,
+    ]
 
 
-class ComponentDetailView(generics.RetrieveAPIView):
+class ComponentDetailView(generics.RetrieveUpdateAPIView):
     """
     Use for read or update endpoints to represent a single model instance.
     Provides get, put, and patch method handlers.
     """
+
     queryset = Component.objects.all()
-    permission_classes = [ComponentPermissions, ]
+    permission_classes = [
+        ComponentPermissions,
+    ]
     serializer_class = ComponentSerializer
 
+    def patch(self, request, *args, **kwargs):
+        return self.partial_update(request, *args, **kwargs)
```

Comment:
This looks suspiciously like the implementation already in `RetrieveUpdateAPIView`. Probably can scrap it unless you want to extend or override it?